doc: improve AsyncLocalStorage sample (#32757)
nodejs/node: doc/api/async_hooks.md

or any other asynchronous duration. It is similar to thread-local storage
in other languages.

> The following example shows how to use `AsyncLocalStorage` to build a simple

Reviewer: ```suggestion
The following example uses `AsyncLocalStorage` to build a simple
```

> logger that assigns ids to incoming HTTP requests and includes them into

Reviewer: Optional nit:

```suggestion
logger that assigns IDs to incoming HTTP requests and includes them into
```

> logger that assigns ids to incoming HTTP requests and includes them into

Reviewer: ```suggestion
logger that assigns ids to incoming HTTP requests and includes them in
```
